sim/battle-actions.ts

```diff
@@ -942,7 +943,13 @@ export class BattleActions {
 		}
 
 		// smartTarget messes up targetsCopy, but smartTarget should in theory ensure that targets will never fail, anyway
-		if (move.smartTarget) targetsCopy = targets.slice(0);
+		if (move.smartTarget) {
```

Comment:
The issue was in this copy, reverting the safety check done for smartTarget earlier